File: 0078-subsets/0078-subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> answer = new ArrayList<>();
        List<Integer> track = new ArrayList<>();

        backTracking(nums,0,track, answer);
        
        return answer;
    }
    public void backTracking(int[] nums, int idx, List<Integer> track, List<List<Integer>> answer) {
        if(idx < nums.length) {
            track.add(nums[idx]);
            backTracking(nums,idx+1,track,answer);
            track.remove(track.size()-1);
            backTracking(nums,idx+1,track,answer);
        }else answer.add(new ArrayList<>(track));
    }
}

File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        def backtrack(curr: List[int], idx: int):
            if idx == len(nums): return

            print(curr)

            while idx < len(nums):
                curr.append(nums[idx])
                res.append(list(curr))

                idx += 1
                backtrack(list(curr), idx)

                curr.pop()
        
        res.append([])
        backtrack([], 0)

        return res